m2m-oauth-server/store/mongodb/tokens.go

```diff
@@ -0,0 +1,172 @@
+package mongodb
+
+import (
+	"context"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-multierror"
+	"github.com/plgd-dev/hub/v2/m2m-oauth-server/pb"
+	"github.com/plgd-dev/hub/v2/m2m-oauth-server/store"
+	"github.com/plgd-dev/hub/v2/pkg/mongodb"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+func (s *Store) CreateToken(ctx context.Context, owner string, token *pb.Token) (*pb.Token, error) {
+	if token.GetOwner() == "" {
+		token.Owner = owner
+	}
+	if token.GetId() == "" {
+		token.Id = uuid.NewString()
+	}
+	if owner != token.GetOwner() {
+		return nil, store.ErrInvalidArgument
+	}
+	err := token.Validate()
+	if err != nil {
+		return nil, err
+	}
+	_, err = s.Store.Collection(tokensCol).InsertOne(ctx, token)
+	if err != nil {
+		return nil, err
+	}
+	return token, nil
+}
+
+func toFilter(owner string, req *pb.GetTokensRequest) (filter bson.D, hint interface{}) {
+	setIdOwnerHint := true
+	if len(req.GetIdFilter()) > 0 {
+		filter = append(filter, bson.E{Key: "_id", Value: bson.M{mongodb.In: req.GetIdFilter()}})
+	} else {
+		setIdOwnerHint = false
+	}
+	if owner != "" {
+		filter = append(filter, bson.E{Key: pb.OwnerKey, Value: owner})
+	} else {
+		setIdOwnerHint = false
+	}
+	if len(req.GetAudienceFilter()) > 0 {
+		filter = append(filter, bson.E{Key: pb.AudienceKey, Value: bson.M{mongodb.In: req.GetAudienceFilter()}})
```

Comment:
add test with audienceFilter (according to SC this code path is not tested)

Reply:
I will remove audience part for now